ranger-core/src/main/java/io/appform/ranger/core/model/ServiceRegistry.java

```diff
@@ -29,6 +29,9 @@ public abstract class ServiceRegistry<T> {
 
     public void updateNodes(List<ServiceNode<T>> nodes) {
         update(nodes);
+    }
+
+    public void markAsRefreshed() {
```

Comment:
Is the `refreshed` variable still used? if not, remove it and remove this method. If yes, explain why you are setting this to true even in failure path... iirc this was used during startup path to ensure the system doesn't start reponding before the system had started. So it would be set to true only once and set like that. In case of initial errors this _should not_ be set to true. So the way forward is this: check if used, if yes, revert this change otherwise remove this variable and the related function completely

Reply:
Changed as per our discussion. Summary:

- For NoNodeException we will return emptyList.
- Any other failure will lead to waiting of the serviceUpdateRegistry
Ideal handling of NoNodeException would be tracked at https://github.com/appform-io/ranger/issues/26